src/commands/dependencies/__tests__/upload.test.ts

```diff
@@ -0,0 +1,405 @@
+    )
+
+    const {context, code} = await runUploadDependenciesCommand(
+      './src/commands/dependencies/__tests__/fixtures/dependencies',
+      {
+        apiKey: 'DD_API_KEY_EXAMPLE',
+        appKey: 'DD_APP_KEY_EXAMPLE',
+        releaseVersion: '1.234',
+        service: 'my-service',
+        source: 'snyk',
+      }
+    )
+
+    const stdout = context.getStdoutBuffer().split(os.EOL)
+    const stderr = context.getStderrBuffer().split(os.EOL)
+
+    expect(stderr).toEqual(['Forbidden'])
+    expect(stdout[6]).toEqual(
+      chalk.red(
+        `Failed upload dependencies: Forbidden. Check ${chalk.bold('DATADOG_API_KEY')} and ${chalk.bold(
+          'DATADOG_APP_KEY'
+        )} environment variables.`
+      )
+    )
+
+    expect(code).toBe(3)
+  })
+})
+
+interface RunUploadDependenciesInput {
+  apiKey?: string
+  appKey?: string
+  dryRun?: boolean
+  releaseVersion?: string
+  service?: string
+  source?: string
+}
+
+async function runUploadDependenciesCommand(filePath: string, input: RunUploadDependenciesInput) {
+  const cli = new Cli()
+  cli.register(UploadCommand)
+
+  const context = createMockContext()
+
+  process.env = {
+    DATADOG_API_KEY: input.apiKey,
+    DATADOG_APP_KEY: input.appKey,
+  }
+  const params = ['dependencies', 'upload', filePath]
+
+  if (input.releaseVersion) {
+    params.push('--release-version', input.releaseVersion)
+  }
+  if (input.service) {
+    params.push('--service', input.service)
+  }
+  if (input.source) {
+    params.push('--source', input.source)
+  }
+  if (input.dryRun) {
+    params.push('--dry-run')
+  }
+
+  const code = await cli.run(params, context)
+
+  return {context, code}
+}
+
+interface MockContext extends BaseContext {
+  getStderrBuffer(): string
+  getStdoutBuffer(): string
+}
+
+const createMockContext = (): MockContext => {
+  const stdoutChunks: any[] = []
+  const stderrChunks: any[] = []
+
+  return {
+    getStderrBuffer: () => stderrChunks.join(''),
+    getStdoutBuffer: () => stdoutChunks.join(''),
+    stderr: new Writable({
+      write(chunk, encoding, callback) {
+        stderrChunks.push(chunk)
+        callback()
+      },
+    }),
+    stdin: new Readable(),
+    stdout: new Writable({
+      write(chunk, encoding, callback) {
+        stdoutChunks.push(chunk)
+        callback()
+      },
+    }),
+  }
+}
+
+function streamToString(stream: Readable): Promise<string> {
+  const chunks: any[] = []
+
+  return new Promise((resolve, reject) => {
+    const handleData = (chunk: any) => chunks.push(chunk)
+    const handleError = (error: any) => {
+      stream.off('data', handleData)
+      stream.off('end', handleEnd)
+
+      reject(error)
+    }
+    const handleEnd = () => {
+      stream.off('data', handleData)
+      stream.off('error', handleError)
+
+      resolve(chunks.join(''))
+    }
+    stream.on('data', handleData)
+    stream.once('error', handleError)
+    stream.once('end', handleEnd)
+
+    stream.resume()
+  })
+}
```

Comment:
Can we move these helper functions outside the test file?

We could have a `helpers.ignore.ts` somewhere in the `__tests__` directory.
And add the pattern `'.+\\.ignore\\.+'` [here](https://github.com/DataDog/datadog-ci/blob/master/jest.config.js#L16).

Reply:
I moved this to the `helpers` directory :)